docs/use_cases/fine_grained_authorization/dynamic.md

```diff
@@ -0,0 +1,416 @@
+);
+
+-- Relationships derived using binary rules.
+declare recursive view derived_binary_relationships (
+    subject_id id_t not null,
+    resource_id id_t not null,
+    relationship id_t
+);
+
+-- All derived relationships.
+declare recursive view derived_relationships (
+    subject_id id_t not null,
+    resource_id id_t not null,
+    relationship id_t
+);
+
+create materialized view derived_unary_relationships as
+select
+    derived_relationships.subject_id,
+    derived_relationships.resource_id,
+    unary_rules.derived_relationship as relationship
+from
+    derived_relationships
+    join unary_rules on derived_relationships.relationship = unary_rules.prerequisite_relationship
+    join objects subject on subject.id = derived_relationships.subject_id
+    join objects resource on resource.id = derived_relationships.resource_id
+where
+    check_condition(unary_rules.condition, subject.properties, resource.properties);
+
+create materialized view derived_binary_relationships as
+select
+    r1.subject_id,
+    r2.resource_id,
+    binary_rules.derived_relationship as relationship
+from
+derived_relationships r1
+    join binary_rules on r1.relationship = binary_rules.prerequisite1_relationship
+    join derived_relationships r2 on r1.resource_id = r2.subject_id and binary_rules.prerequisite2_relationship = r2.relationship
+    join objects subject on subject.id = r1.subject_id
+    join objects resource on resource.id = r2.resource_id
+where
+    check_condition(binary_rules.condition, subject.properties, resource.properties);
+
+create materialized view derived_relationships as
+select * from relationships
+UNION ALL
+select * from derived_unary_relationships
+UNION ALL
+select * from derived_binary_relationships;
+```
+
+### Lights, Camera, Action!
+
+Let's see if it works. Open the complete code provided above in one of the following environments:
+* Feldera online sandbox: [https://try.feldera.com/create/?name=dynamic-fga](https://try.feldera.com/create/?name=dynamic-fga) OR
+* Your local Feldera instance: [localhost:8080/create/?name=dynamic-fga](http://localhost:8080/create/?name=dynamic-fga)
+
+Start the pipeline and create rules for the file manager example using ad hoc queries:
+
+```sql
+insert into unary_rules values
+  ('editor', '`true`', 'group-can-write'),         -- Rule 1.
+  ('viewer', '`true`', 'group-can-read'),          -- Rule 3.
+  ('group-can-write', '`true`', 'group-can-read'); -- Rule 4.
+
+insert into binary_rules values
+  ('group-can-write', 'parent', '`true`', 'group-can-write'),                     -- Rule 2.
+  ('group-can-read', 'parent', '`true`', 'group-can-read'),                       -- Rule 5.
+  ('member', 'group-can-write', 'subject.is_banned != `true`', 'user-can-write'), -- Rule 6.
+  ('member', 'group-can-read', 'subject.is_banned != `true`', 'user-can-read');   -- Rule 7.
+```
+
+Populate the object graph:
+
+```sql
+insert into objects values
+  ('user:emily', '{"is_banned": false}'),
```

Comment:
there is some kind of namespace encoded in these strings with these prefixes

Reply:
It's purely for convenience. I thought that 'user:emily', 'group:it' is easier to follow than 'emily', 'it'. We don't do anything special about these prefixes.